src/python/pants/backend/docker/tailor_test.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.backend.docker.tailor import PutativeDockerTargetsRequest
+from pants.backend.docker.tailor import rules as docker_tailor_rules
+from pants.backend.docker.target_types import DockerImage
+from pants.core.goals.tailor import (
+    AllOwnedSources,
+    PutativeTarget,
+    PutativeTargets,
+    PutativeTargetsSearchPaths,
+)
+from pants.engine.rules import QueryRule
+from pants.testutil.rule_runner import RuleRunner
+
+
+def test_find_putative_targets() -> None:
+    rule_runner = RuleRunner(
+        rules=[
+            *docker_tailor_rules(),
+            QueryRule(PutativeTargets, [PutativeDockerTargetsRequest, AllOwnedSources]),
+        ],
+        target_types=[
+            DockerImage,
+        ],
+    )
+    rule_runner.create_file("src/docker_ok/Dockerfile")
+    rule_runner.create_file("src/docker_orphan/Dockerfile")
```

Comment:
I've been trying to actively kill this old util in favor of `.write_files()`:

```suggestion
    rule_runner.write_files({"src/docker_ok/Dockerfile": "", "src/docker_orphan/Dockerfile": ""})
```

```python

Reply:
maybe add `@deprecated` on `RuleRunner.create_file` ?

Reply:
I'd like to, but we have to first finish updating all call sites. I think we're about 60% there :)

Reply:
Why? doesn't deprecated just warn in the logs? until you hit the removal version, which could be bumped until all call sites have migrated away..

Reply:
True. That is a bit noisy for all those tests, but maybe the noise is motivating. 

I'm headed to bed - feel free to put up a deprecation if you'd like :)

Reply:
Issue #12441

Reply:
I think we're getting a little distracted here. There is no urgent need to deprecate this, and I don't see why we should bother.